File: source/hgsys/domain/edit_mode.py
from enum import IntEnum


class EditMode(IntEnum):
    NONE = 0
    APPEND = 1
    MODIFY = 2
    INHIBIT = 3


_EDITABLE = {
    EditMode.NONE: False,
    EditMode.APPEND: True,
    EditMode.MODIFY: True,
    EditMode.INHIBIT: False,
}

_COLOR = {
    EditMode.NONE: "black",
    EditMode.APPEND: "red",
    EditMode.MODIFY: "red",
    EditMode.INHIBIT: "black",
}


def is_editable(mode: EditMode) -> bool:
    return _EDITABLE[mode]


def color_for(mode: EditMode) -> str:
    return _COLOR[mode]

File: frontend/src/lib/editMode.ts
// 對應 pkg/domain.EditMode, 以 string union 的形式保留, 提升可讀性.

export type EditMode = "none" | "append" | "modify" | "inhibit";

export function isEditable(m: EditMode): boolean {
  return m === "append" || m === "modify";
}

// CSS class 提示: 可編輯模式時欄位以紅色呈現, 與舊版程式風格一致.
export function editClass(m: EditMode): string {
  return isEditable(m) ? "edit-red" : "";
}
